asserts/constraint.go

```diff
@@ -218,15 +219,23 @@ var (
 )
 
 func compileEvalAttrMatcher(cc compileContext, s string) (attrMatcher, error) {
-	if len(cc.opts.allowedOperations) == 0 {
-		return nil, fmt.Errorf("cannot compile %q constraint %q: no $OP() constraints supported", cc, s)
+	if len(cc.opts.allowedOperations) == 0 && len(cc.opts.allowedRefs) == 0 {
+		return nil, fmt.Errorf("cannot compile %q constraint %q: no $OP() or $REF constraints supported", cc, s)
 	}
+
+	if strutil.ListContains(cc.opts.allowedRefs, s[1:]) {
+		return evalAttrMatcher{op: s[1:]}, nil
+	}
+
 	ops := validEvalAttrMatcher.FindStringSubmatch(s)
```

Comment:
Could maybe use a comment explaining how the `strutil.ListContains` early return case differs from the `FindStringSubmatch` path.

Reply:
In particular, the regexp doesn't match `$PLUG_PUBLISHER_ID()`, so everything after the early return should not include any of the allowed refs, right?